src/util/LazyJsonParser.cpp

```diff
@@ -22,12 +22,14 @@ LazyJsonParser::Generator LazyJsonParser::parse(
   Details& details = co_await cppcoro::getDetails;
   for (const auto& chunk : partialJson) {
     if (details.first100_.size() < 100) {
-      details.first100_ += chunk.substr(0, 100 - details.first100_.size());
+      details.first100_ += chunk.substr(0, 100);
     }
     if (auto res = p.parseChunk(chunk); res.has_value()) {
       co_yield res;
 
       if (p.endReached_) {
+        details.last100_ = chunk.substr(
+            std::max(0, static_cast<int>(chunk.size() - 100)), 100);
```

Comment:
I think we want to have the last 100 always. 
So overwrite them directly after setting the `first100`, but outside the if of course.
That way we also get the context if the parsing of the chunk failed, which is probably the most useful.